src/components/AddVehicleForm.jsx

```diff
@@ -0,0 +1,117 @@
+      toast.success('Vehicle added successfully');
+    } catch (error) {
+      console.error(error.message);
+    }
+  };
+
+  return (
+    <form className="w-full md:w-1/3" onSubmit={handleVehicleSubmit}>
+      {/* eslint-disable jsx-a11y/label-has-associated-control */}
+      <div className="flex flex-col gap-3">
+        <p className="text-4xl text-center">Create Car</p>
+        <div className="form-control">
+          <label className="input-group input-group-vertical">
+            <span className="font-bold">Car Name</span>
+            <input
+              type="text"
+              name="name"
+              className="input input-bordered focus:outline-none"
+              value={name}
+              onChange={(e) => setName(e.target.value)}
+            />
+          </label>
+        </div>
+        <div className="form-control">
+          <label className="input-group input-group-vertical">
+            <span className="font-bold">Model</span>
+            <input
+              type="text"
+              name="model"
+              className="input input-bordered focus:outline-none"
+              value={model}
+              onChange={(e) => setModel(e.target.value)} // Corrected onChange handler
+            />
+          </label>
+        </div>
+        <div className="form-control">
+          <label className="input-group input-group-vertical">
+            <span className="font-bold">Price ($)</span>
+            <input
+              type="number"
+              name="price"
+              className="input input-bordered focus:outline-none"
+              value={price}
+              onChange={(e) => setPrice(e.target.value)} // Corrected onChange handler
+            />
+          </label>
+        </div>
+        <div className="form-control">
+          <label className="input-group input-group-vertical">
+            <span className="font-bold">Photo</span>
+            <input
+              type="text"
+              name="photo"
+              className="input input-bordered focus:outline-none"
+              value={photo}
+              onChange={(e) => setPhoto(e.target.value)} // Corrected onChange handler
+            />
+          </label>
+        </div>
+        <div className="form-control">
+          <label className="input-group input-group-vertical">
+            <span className="font-bold">Description</span>
+            <textarea
+              type="text"
+              name="description"
+              className="input input-bordered focus:outline-none"
+              value={description}
+              onChange={(e) => setDescription(e.target.value)}
+            />
+          </label>
+        </div>
+        <div className="form-control">
+          <button type="submit" className="btn btn-primary" disabled={isLoading}>
+            {isLoading ? 'Adding...' : 'Add Car'}
+          </button>
+        </div>
+      </div>
+    </form>
+  );
+};
+
+export default AddVehicleForm;
```

Comment:
Your AddVehicleForm component looks generally good, but I do have a few recommendations for you:

- [ ] **_Form Accessibility:_** As a best practice, you should associate your labels with their corresponding inputs using the htmlFor attribute on the label element. This ensures better accessibility for screen readers and improves usability.

- [ ]  **Form Validation:** Consider adding form validation to ensure that the data entered by the user is valid before submitting the form. For example, you can check if the required fields are filled, and if the price is a positive number.

- [ ]  **Error Handling:** In the handleVehicleSubmit function, you have a try-catch block to handle errors when dispatching the postVehicle action. It's a good approach, but consider adding feedback to the user when an error occurs, so they know something went wrong.